Firebase/InstanceID/FIRInstanceID.m

```diff
@@ -738,29 +717,20 @@ - (BOOL)isFCMAutoInitEnabled {
     return NO;
   }
 
-  // Messaging doesn't have the singleton method, auto init should be enabled since FCM exists.
-  SEL instanceSelector = NSSelectorFromString(kFIRInstanceIDFCMSelectorInstance);
-  if (![messagingClass respondsToSelector:instanceSelector]) {
-    return YES;
-  }
-
-  // Get FIRMessaging shared instance.
-  IMP messagingInstanceIMP = [messagingClass methodForSelector:instanceSelector];
-  id (*getMessagingInstance)(id, SEL) = (void *)messagingInstanceIMP;
-  id messagingInstance = getMessagingInstance(messagingClass, instanceSelector);
-
-  // Messaging doesn't have the property, auto init should be enabled since FCM exists.
+  // Messaging doesn't have the class method, auto init should be enabled since FCM exists.
   SEL autoInitSelector = NSSelectorFromString(kFIRInstanceIDFCMSelectorAutoInitEnabled);
-  if (![messagingInstance respondsToSelector:autoInitSelector]) {
+  if (![messagingClass respondsToSelector:autoInitSelector]) {
     return YES;
   }
 
-  // Get autoInitEnabled method.
-  IMP isAutoInitEnabledIMP = [messagingInstance methodForSelector:autoInitSelector];
-  BOOL (*isAutoInitEnabled)(id, SEL) = (BOOL(*)(id, SEL))isAutoInitEnabledIMP;
+  // Get the autoInitEnabled class method.
+  IMP isAutoInitEnabledIMP = [messagingClass methodForSelector:autoInitSelector];
+  BOOL(*isAutoInitEnabled)
```

Comment:
I believe you can cast through `(void *)` to avoid repeating yourself on the right hand side of this statement.